test/phoenix/router/helpers_test.exs

```diff
@@ -152,6 +152,22 @@ defmodule Phoenix.Router.HelpersTest do
     end
   end
 
+  test "top-level named routes with complex ids" do
+    assert Helpers.post_path(__MODULE__, :show, "8==D--+") == 
```

Comment:
The choice of characters `8==D` do not really belong to a technical codebase. If you haven't noticed it, it is OK because I haven't either the first time I reviewed it. Can we pick something else? Dropping the number 8 would be fine.